packages/cubejs-api-gateway/src/gateway.ts

```diff
@@ -1040,26 +1040,73 @@ class ApiGateway {
     const requestStarted = new Date();
     try {
       if (!query) {
-        throw new UserError(
-          'A user\'s query must contain a body'
-        );
+        throw new UserError('A user\'s query must contain a body');
       }
 
       if (!Array.isArray(query) && !query.query) {
         throw new UserError(
           'A user\'s query must contain at least one query param.'
         );
       }
-
+      
       query = {
         ...query,
-        requestId: context.requestId
+        requestId: context.requestId,
       };
 
-      if (query.resultFilter?.objectTypes && !Array.isArray(query.resultFilter.objectTypes)) {
-        throw new UserError(
-          'A query.resultFilter.objectTypes must be an array of strings'
-        );
+      // Determine SQL query type and add LIMIT clause if needed
+      const shouldAddLimit = (sql: string): Boolean => {
```

Comment:
Could you please move this helper function out from the method scope to a separate file in the `helpers`?